### PR TITLE
[AMSDK-8908] Add tutorial for setting up rules with target events and attach data

### PR DESCRIPTION
**Pull request by rymorale**

### CODE REVIEW

File: resources/user-guides/attach-data.md [hunk 1]
```diff
@@ -107,3 +110,67 @@ After you complete your configuration, verify that your rule looks like the foll
 1. Click **Save**
 2. Rebuild your Launch property and deploy it to the correct Environment.
 
+{% endtab %}
+
+{% tab title="Target" %}
+
+The following sample shows how to _attach data_ to all outgoing `loadRequests` or `retrieveLocationContent` Target network requests. To create this type of rule, select your property in Experience Platform Launch and complete the following steps:
```
Comment (swarna04) [inline resources/user-guides/attach-data.md]: Let us not mention the deprecated API `loadRequests`

**Comment by swarna04** [comment]
The tutorial looks fine @rymorale. The main reason we wanted this tutorial was because Target event formats for prefetch/ load requests are quite complicated. This tutorial would give customers a JSON payload reference to quickly setup attach data for Target. I have two major feedback points here:
1. Provide example for adding parameters to the Target objects or adding new Target objects in prefetch or request array.  
2. Cover examples for all the four Target event types exposed in Launch.

File: resources/user-guides/attach-data.md [hunk 2]
```diff
@@ -107,3 +110,271 @@ After you complete your configuration, verify that your rule looks like the foll
...
+
+1. Under the **Actions** section, click **Add**.
+2. From the **Extension** drop-down list, select **Mobile Core**.
+3. From the **Action Type** drop-down list, select **Attach Data**.
+4. On the right pane, in the **JSON Payload** field, type the data that will be added to this event.
+5. Click **Keep Changes**.
+
+On the right pane, you can add a freeform JSON payload that adds data to an SDK event before an extension that is listening for this event can hear the event. In this example, a custom mbox with custom mbox parameters is added to the event before the Target extension processes it. The added custom mbox will now be added on outgoing Target retrieve location requests.
+
+In the following example, an mbox named **custom_mbox** with an additional mbox parameter named **customparams** is added to the Target event.
```
Comment (swarna04) [inline resources/user-guides/attach-data.md]: We should mention - In the following example, an additional `TargetRequest` object for the given mbox `custom_mbox` containing the provided Target parameters is added to the Target event.

File: resources/user-guides/attach-data.md [hunk 3]
```diff
@@ -107,3 +110,271 @@ After you complete your configuration, verify that your rule looks like the foll
...
+
+1. Under the **Actions** section, click **Add**.
+2. From the **Extension** drop-down list, select **Mobile Core**.
+3. From the **Action Type** drop-down list, select **Attach Data**.
+4. On the right pane, in the **JSON Payload** field, type the data that will be added to this event.
+5. Click **Keep Changes**.
+
+On the right pane, you can add a freeform JSON payload that adds data to an SDK event before an extension that is listening for this event can hear the event. In this example, a custom mbox with custom mbox parameters is added to the event before the Target extension processes it. The added custom mbox will now be added on outgoing Target prefetch requests.
+
+In the following example, an mbox named **custom_mbox** with an additional mbox parameter named **customparams** is added to the Target event.
```
Comment (swarna04) [inline resources/user-guides/attach-data.md]: We should mention - In the following example, an additional `TargetPrefetch` object for the given mbox `custom_mbox` containing the provided Target parameters is added to the Target event.

**Comment by swarna04** [comment]
@rekrishn Can you please take a look at this PR again, we decided to add additional content here.

File: resources/user-guides/attach-data.md [hunk 4]
```diff
@@ -114,7 +114,7 @@ After you complete your configuration, verify that your rule looks like the foll
 
 {% tab title="Target" %}
 
-### How to attach additional profile parameters to Target retrieve location events
+### How to attach additional Target request objects to Target retrieve location events
```
Comment (rekrishn) [inline resources/user-guides/attach-data.md]: Please reword to:
"Attaching additional Target request objects to Target to retrieve location events"

File: resources/user-guides/attach-data.md (hunk 2 above)
Comment (rekrishn) [reply resources/user-guides/attach-data.md]: @rymorale, please reword Swarna's suggested text as follows:
"In the following example, an additional TargetRequest object for the given `custom_mbox`, which contains the provided Target parameters, is added to the Target event.

File: resources/user-guides/attach-data.md [hunk 5]
```diff
@@ -155,25 +155,26 @@ If you do not have existing rules for this property, the **Create New Rule** but
...
+In the following example, an mbox named **custom_mbox** with an additional mbox parameter named **customparams** is added to the Target event.
 
 ![](../../.gitbook/assets/target-attach-data-json-example.png)
 
-The following example shows how the data element for OS version was created.
-
-![](../../.gitbook/assets/target-attach-data-data-element-setup.png)
-
-In addition to being able to add custom profile parameters to each Target retrieve location request, custom mboxes can be added as well. The following example contains a valid JSON payload for this use case:
+In the above example, the JSON payload adds custom mbox parameters only for the custom mbox added. Custom parameters can be added to each of the Target retrieve location objects instead. The following example contains a valid JSON payload for this use case:
```
Comment (rekrishn) [inline resources/user-guides/attach-data.md]: Please reword as follows:
"In the above example, the JSON payload adds custom mbox parameters only for the custom mbox that was added. You can also add custom parameters to each of the Target retrieve location objects. The following example contains a valid JSON payload for this use case:"

File: resources/user-guides/attach-data.md [hunk 6]
```diff
@@ -233,7 +234,7 @@ In the following example, an mbox named **custom_mbox** with an additional mbox
 
 ![](../../.gitbook/assets/target-attach-data-json-example-prefetch.png)
 
-In the above example, the JSON payload adds custom mbox parameters only for the custom mbox added. Custom parameters can be added to each of the Target objects instead. The following example contains a valid JSON payload for this use case:
+In the above example, the JSON payload adds custom mbox parameters only for the custom mbox added. Custom parameters can be added to each of the Target prefetch objects instead. The following example contains a valid JSON payload for this use case:
```
Comment (rekrishn) [inline resources/user-guides/attach-data.md]: Please reword as follows:
"In the above example, the JSON payload adds custom mbox parameters only for the custom mbox that was added. You can also add custom parameters to each of the Target retrieve prefetch objects. The following example contains a valid JSON payload for this use case:"

File: resources/user-guides/attach-data.md [hunk 7]
```diff
@@ -259,7 +260,7 @@ After you complete your configuration, verify that your rule looks like the foll
 1. Click **Save**
 2. Rebuild your Launch property and deploy it to the correct Environment.
 
-### How to attach additional mbox and profile parameters to Target location clicked events
+### How to attach additional Target parameters to Target location clicked events
```
Comment (rekrishn) [inline resources/user-guides/attach-data.md]: Please reword to:
"Attaching additional Target parameters to Target location-clicked events"

File: resources/user-guides/attach-data.md [hunk 8]
```diff
@@ -302,10 +303,14 @@ If you do not have existing rules for this property, the **Create New Rule** but
 
 On the right pane, you can add a freeform JSON payload that adds data to an SDK event before an extension that is listening for this event can hear the event. In this example, custom mbox and profile parameters are added to the event before the Target extension processes it. The added mbox and profile parameters will now be added on outgoing Target location clicked requests.
 
-In the following example, **extraKey**, **extraKey2**, and a data element defined for the **OS version** are added to the profile parameters of the Target event. A key named **customMboxParameter** is added to the mbox parameters of the Target event. Values for the new keys can either be hardcoded in the rule, or dynamically determined by the SDK when this event processes by using Data Elements.
+In the following example, **extraKey** and **extraKey2** are added to the profile parameters. A key named **customMboxParameter** and a data element defined for the **OS version** are added to the mbox parameters of the Target event. Values for the new keys can either be hardcoded in the rule, or dynamically determined by the SDK when this event processes by using Data Elements.
```
Comment (rekrishn) [inline resources/user-guides/attach-data.md]: - In the 2nd sentence, please add "that was" between "element" and "defined. 
- In the 3rd sentence, please add "be" between "or" and "dynamically" and remove the comma between "rule" and "or"
- In the 3rd sentence, please put "Data Elements" in lower case --> "data elements"

File: resources/user-guides/attach-data.md [hunk 9]
```diff
@@ -302,10 +303,14 @@ If you do not have existing rules for this property, the **Create New Rule** but
...
 
 ![](../../.gitbook/assets/target-attach-data-json-example-location-clicked.png)
 
+The following example shows how the data element for OS version was created.
```
Comment (rekrishn) [inline resources/user-guides/attach-data.md]: Please add "this" between "for" and "OS"

File: resources/user-guides/attach-data.md [hunk 10]
```diff
@@ -315,7 +320,7 @@ After you complete your configuration, verify that your rule looks like the foll
 1. Click **Save**
 2. Rebuild your Launch property and deploy it to the correct Environment.
 
-### How to attach additional mbox and profile parameters to Target location displayed events
+### How to attach additional Target parameters to Target location displayed events
```
Comment (rekrishn) [inline resources/user-guides/attach-data.md]: Please reword to:
"Attaching additional Target parameters to Target location-displayed events"

File: resources/user-guides/attach-data.md [hunk 11]
```diff
@@ -358,7 +363,7 @@ If you do not have existing rules for this property, the **Create New Rule** but
 
 On the right pane, you can add a freeform JSON payload that adds data to an SDK event before an extension that is listening for this event can hear the event. In this example, custom mbox and profile parameters are added to the event before the Target extension processes it. The added mbox and profile parameters will now be added on outgoing Target location displayed requests.
 
-In the following example, **extraKey**, **extraKey2**, and a data element defined for the **OS version** are added to the profile parameters of the Target event. A key named **customMboxParameter** is added to the mbox paramters of the Target event. Values for the new keys can either be hardcoded in the rule, or dynamically determined by the SDK when this event processes by using Data Elements.
+In the following example, **extraKey** and **extraKey2** are added to the profile parameters. A key named **customMboxParameter** and a data element defined for the **OS version** are added to the mbox parameters of the Target event. Values for the new keys can either be hardcoded in the rule, or dynamically determined by the SDK when this event processes by using Data Elements.
```
Comment (rekrishn) [inline resources/user-guides/attach-data.md]: - In the 2nd sentence, please add "that was" between "element" and "defined. 
- In the 3rd sentence, please add "be" between "or" and "dynamically" and remove the comma between "rule" and "or"
- In the 3rd sentence, please put "Data Elements" in lower case --> "data elements"

File: resources/user-guides/attach-data.md (hunk 6 above)
Comment (rymorale) [reply resources/user-guides/attach-data.md]: I've updated to this suggestion but left out the word `retrieve` as it doesn't apply to prefetch.

**Pull request merged**